r-package/grf/R/input_utilities.R

```diff
@@ -191,21 +191,33 @@ validate_newdata <- function(newdata, X) {
   validate_X(newdata)
 }
 
+validate_sample_weights = function(sample.weights, X) { 
+    if(!is.null(sample.weights)) {
+        if(length(sample.weights) != nrow(X)) { 
+            stop("sample.weights has incorrect length")
+        }
+        if(any(sample.weights < 0)) {
+            stop("sample.weights must be nonnegative")
+        }
+    }
+}
+
 coerce_honesty_fraction <- function(honesty.fraction) {
   if(is.null(honesty.fraction)) {
     return(0)
   }
   honesty.fraction
 }
 
-create_data_matrices <- function(X, ...) {
+create_data_matrices <- function(X, ..., sample.weights=NULL) {
```

Comment:
I don't think this named argument `sample.weights` is ever used. From my testing, the `...` is just capturing `sample.weights` everywhere it's passed in. This works because `cbind` can handle a `NULL` argument.

I would be fine just removing this named argument, and relying on this behavior of `cbind`. I checked this works for both the normal and sparse versions of `cbind`.

Reply:
It's used if you pass by keyword, which I apparently intended to do but did not. Preference for passing by keyword or getting rid of the keyword argument?

Reply:
The new version passes by keyword, but doesn't do any such preprocessing.

Reply:
👍 works for me, as long as we're actually using the keyword argument.